### PR TITLE
[FEATURE] Consent Management Package

### PR DESCRIPTION
**Pull request by jdamner**
New Feature 🎉 

Consent management, built on top of https://github.com/orestbida/cookieconsent, this package provides a simple interface in WP to configure and use the consent management solution combined with GTM consent modes.

### CODE REVIEW

File: packages/query-include-exclude/package.json [hunk 1]
```diff
@@ -15,7 +15,7 @@
 		"test:snapshots": "wp-scripts test-unit-js --updateSnapshot",
 		"packages-update": "wp-scripts packages-update --dist-tag=wp-6.4",
 		"start": "wp-scripts start",
-		"type-coverage": "type-coverage --strict --detail --at-least 97.5"
+		"type-coverage": "type-coverage --strict --detail --at-least 92.03"
```
Comment (Ritchbox) [inline packages/query-include-exclude/package.json:18]: Is this meant to be included?

Comment (jdamner) [reply packages/query-include-exclude/package.json:18]: Yeah, one of the WP packages somehow lost its type data and now the type coverage on that package is reduced. Will look into it in the future.

File: packages/consent-management/consent-management.php [hunk 2]
```diff
@@ -0,0 +1,172 @@
+<?php
+/**
+ * Plugin Name: ConsentManagement
+ * Description: A brief description of the plugin.
```
Comment (Ritchbox) [inline packages/consent-management/consent-management.php:4]: Needs updating

Comment (jdamner) [reply packages/consent-management/consent-management.php:4]: I always miss that... I'll have to update the create-package script to make sure I don't forget.

File: packages/consent-management/consent-management.php [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+	 */
+	private const OPTION_NAME = 'consent_options';
+
+	/**
+	 * Init Hooks
+	 *
+	 * @return void
+	 */
+	public function init(): void {
+		add_action( 'wp_enqueue_scripts', array( $this, 'enqueue_frontend' ) );
```
Comment (Ritchbox) [inline packages/consent-management/consent-management.php:41]: Long form arrays? Not opposed to it but thought we didn't use them

Comment (jdamner) [reply packages/consent-management/consent-management.php:41]: I think it's default to long form in WPCS. I've not modified the default wpcs config tbh, I'm keen for a fairly generic/standard that's not so opinionated by box ( even if opinionated by WP ).

File: packages/consent-management/consent-management.php [hunk 4]
```diff
@@ -0,0 +1,172 @@
...
+	public function render_settings_screen(): void {
+		?>
+		<div class='wrap'>
+			<div id="consent-settings-root"></div>
+		</div>
+		<?php
+	}
+}
+
+add_action(
```
[the remark this answers is not in the document]
Comment (jdamner) [reply packages/consent-management/consent-management.php:166]: I know, fairly common practice in WP though. I should move it out but then I've got to setup auto loading or dependency etc, where this class isn't exactly complicated. I could've made it namespaced or anonymous functions if you're opposed to the class in the root file approach.

**Review by craigpearson — APPROVED** [review]

File: packages/consent-management/consent-management.php (hunk 3 above)
Comment (Ritchbox) [reply packages/consent-management/consent-management.php:41]: Yeah there was a couple other things I noticed but all good for keeping it as generic as possible

**Review by Ritchbox — APPROVED** [review]

File: packages/consent-management/consent-management.php (hunk 2 above)
Comment (jdamner) [reply packages/consent-management/consent-management.php]: Updated, and added a line in the create-package scripts as a basic reminder for now. Someone like @craigpearson who's a whiz will end up making some amazing script that basically does all the work for you...

**Pull request merged**